e2e/_suites/fleet/fleet.go

```diff
@@ -312,102 +301,70 @@ func (fts *FleetTestSuite) anAgentIsDeployedToFleetWithInstaller(image string, i
 }
 
 func (fts *FleetTestSuite) anAgentIsDeployedToFleetWithInstallerAndFleetServer(image string, installerType string) error {
+	service := fmt.Sprintf("%s-systemd", image)
 	log.WithFields(log.Fields{
 		"image":     image,
+		"service":   service,
 		"installer": installerType,
 	}).Trace("Deploying an agent to Fleet with base image using an already bootstrapped Fleet Server")
 
 	fts.Image = image
-	fts.InstallerType = installerType
 
-	agentInstaller := fts.getInstaller()
-
-	containerName := fts.getContainerName(agentInstaller, 1) // name of the container
+	if strings.EqualFold(fts.Image, "debian") && installerType == "systemd" {
```

Comment:
If the feature file is passing the `deb/rpm` installers, is this still needed? I fought this too in #1162

Reply:
With your changes in #1162 we dont need this any longer, since that information will already be known. The more abstraction we can place in the service installer the less we need to be explicit about in the fleet test suite